lib/Parse/Lexer.cpp

```diff
@@ -862,8 +862,12 @@ void Lexer::lexHexNumber() {
   if (*CurPtr != '.' && *CurPtr != 'p' && *CurPtr != 'P')
     return formToken(tok::integer_literal, TokStart);
   
+  const char *PtrOnDot = nullptr;
+  bool letterAfterDot = false;
+
   // (\.[0-9A-Fa-f][0-9A-Fa-f_]*)?
   if (*CurPtr == '.') {
+    PtrOnDot = CurPtr;
```

Comment:
Please sink the declaration of PtrOnDot to this line, since it is only used within this if statement, and therefore is always valid (never null).